Xcode 11

foulkesjohn opened the pull request.

Commit:
Updated Swift 5 + SwiftPM

Commit:
Updated for xcode 11

babbage commented:
I have separated out the Swift 5 migration and am committing that as a initial pull request https://github.com/Antondomashnev/FBSnapshotsViewer/pull/82

babbage committed:
Merge branch 'master' into xcode_11

babbage commented:
There is some work here that appears to be potentially useful, but there are a number of issues with this pull request. I am unclear how you are using xcparse in this pull request. You have `import xcparse` in MenuInteractor.swift but xcparse is a command line utility, not a Swift module, so at least as set up in in the pull request, it cannot be imported. The XCParseCore library does not expose the XCPParse class that you utilise in your code. The branch does not build on my machine. It would be helpful to me if you could step me through how you configured this in your Xcode project as it presumably was working for you?

Please also verify you followed the steps in the [Contributor Guide](https://github.com/Antondomashnev/FBSnapshotsViewer/blob/master/CONTRIBUTING.md) including that the branch passed all tests when running `bundle exec fastlane mac test`.

Finally, this pull request also introduces a number of unnecessary files, including a second FBSnapshotsViewer.xcproj in /Sources and boilerplate files in the /Tests directory that was introduced.

babbage commented:
By the way, I recommend keeping Fastlane on v 2.137.0, as the current 2.141.0 appears to have an issues that needs fixing in Fastlane, that prevents our tests from running.

foulkesjohn commented:
It's been a while since I looked at this as I did it a test to see what was possible. I added XCParse by cloning the repo and adding the project to this xcode proj. This was due to XCParse being a swift package and not a cocoapod. XCParse requires the full bundle to be present before it can be parsed which meant the time that snapshots were available to compare is very long. It's pretty much unusable for me and I've moved onto use kaleidoscope, which is a shame. I pushed it up incase anyone wants to take this and see if they can do anything better.
